commit 05c6adf8104b4321d3a3716a7b9feb6bf223ed0c (HEAD, nixpkgs)
Author: Vladimír Čunát <vcunat@gmail.com>
Date:   Tue Nov 4 12:24:25 2014 +0100

    add check for /etc/fonts/@configVersion@/fonts.conf
    
    It's checked between FONTCONFIG_FILE and the usual /etc/fonts/fonts.conf.
    Also, hardcode /etc/fonts/fonts.conf to prevent accidental override.

diff --git a/src/fccfg.c b/src/fccfg.c
index 6377fd7..e9eb10a 100644
--- a/src/fccfg.c
+++ b/src/fccfg.c
@@ -2070,8 +2070,13 @@ FcConfigFilename (const FcChar8 *url)
     if (!url || !*url)
     {
 	url = (FcChar8 *) getenv ("FONTCONFIG_FILE");
+	if (!url) {
+	    static const FcChar8 *cfPath = "/etc/fonts/@configVersion@/fonts.conf";
+	    if (access (cfPath, R_OK) == 0)
+		url = cfPath;
+	}
 	if (!url)
-	    url = (FcChar8 *) FONTCONFIG_FILE;
+	    url = (FcChar8 *) "/etc/fonts/fonts.conf";
     }
     file = 0;
 
